src/Components/Blazor/testassets/Wasm.Authentication.Client/Shared/LoginDisplay.razor

```diff
@@ -0,0 +1,21 @@
+@using Microsoft.AspNetCore.Components.Authorization
+@inject NavigationManager Navigation
+@inject SignOutSessionStateManager SignOutManager
+<AuthorizeView>
+    <Authorized>
+        <a href="authentication/profile">Hello, @context.User.Identity.Name!</a>
+        <button class="nav-link btn btn-link" @onclick="BeginLogout">Log out</button>
+    </Authorized>
+    <NotAuthorized>
+        <a href="authentication/register">Register</a>
+        <a href="authentication/login">Log in</a>
+    </NotAuthorized>
+</AuthorizeView>
+
+@code{
+    private async Task BeginLogout(MouseEventArgs args)
+    {
+        await SignOutManager.SetSignOutState();
+        Navigation.NavigateTo("authentication/logout");
```

Comment:
Is this the API we expect customers to use? It seems more complicated than I'd expect. (1) you need to know about `SignOutManager.SetSignOutState` which is not obvious, and (2) you have to hardcode this logout URL which the system is already meant to know about anyway.

Instead of this, how about getting the developer to inject the `IRemoteAuthenticationService` and call `SignOutAsync` on that? It could deal with tracking any internal CSRF state as well as issuing the redirection.

If that's already possible, perhaps we could eliminate `SignOutManager` from the public API surface, so there's only one set of APIs for this.

Reply:
Discussed offline, going with the current approach.